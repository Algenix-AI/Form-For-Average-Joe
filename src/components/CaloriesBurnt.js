import { Typography, Card, Grid } from '@mui/material';

const CaloriesBurnt = ({ cal }) => {
    //todo fix levelData formula
    const levelData = cal > 5000 ? 0 : 1000 - (cal % 1000);

    return (
        <Card sx={{paddingTop: "1rem"}}>
            <Grid container justifyContent="center" alignItems="center" direction="column">
                <Grid item style={{ paddingTop: "1rem" }}>
                    <Typography variant="subtitle1" align="center">
                        Cumulative Calories Burnt:
                    </Typography>
                    <Typography variant="h2" align="center">
                        {cal}
                    </Typography>
                </Grid>
                <Grid item style={{ paddingTop: "2rem" }}>
                    <Typography variant="subtitle1" align="center">
                        Calories required to reach next level:
                    </Typography>
                    <Typography variant="h2" align="center">
                        {levelData}
                    </Typography>
                </Grid>
            </Grid>
        </Card>
    );
}

export default CaloriesBurnt;