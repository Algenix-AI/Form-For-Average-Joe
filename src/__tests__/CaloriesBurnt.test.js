import { render, screen } from '@testing-library/react';
import CaloriesBurnt from '../components/CaloriesBurnt';

test('renders react component', () => {
    render(<CaloriesBurnt cal={1402} />);
    const currentCal = screen.getByText("1402.0");
    const requiredCal = screen.getByText("598")
    expect(currentCal).toBeInTheDocument();
    expect(requiredCal).toBeInTheDocument();
});

test('renders react component', () => {
    render(<CaloriesBurnt cal={0} />);
    const currentCal = screen.getByText("0");
    const requiredCal = screen.getByText("1000")
    expect(currentCal).toBeInTheDocument();
    expect(requiredCal).toBeInTheDocument();
});

test('renders react component', () => {
    render(<CaloriesBurnt cal={5882} />);
    const currentCal = screen.getByText("5882.0");
    const requiredCal = screen.getByText("0")
    expect(currentCal).toBeInTheDocument();
    expect(requiredCal).toBeInTheDocument();
});