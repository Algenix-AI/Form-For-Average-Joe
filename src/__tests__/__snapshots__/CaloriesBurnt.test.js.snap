// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders react component 1`] = `
Object {
  "currentCal": <h2
    class="MuiTypography-root MuiTypography-h2 MuiTypography-alignCenter css-icizyq-MuiTypography-root"
  >
    1402
  </h2>,
  "requiredCal": <h2
    class="MuiTypography-root MuiTypography-h2 MuiTypography-alignCenter css-icizyq-MuiTypography-root"
  >
    598
  </h2>,
}
`;

exports[`renders react component 2`] = `
Object {
  "currentCal": <h2
    class="MuiTypography-root MuiTypography-h2 MuiTypography-alignCenter css-icizyq-MuiTypography-root"
  >
    0
  </h2>,
  "requiredCal": <h2
    class="MuiTypography-root MuiTypography-h2 MuiTypography-alignCenter css-icizyq-MuiTypography-root"
  >
    1000
  </h2>,
}
`;

exports[`renders react component 3`] = `
Object {
  "currentCal": <h2
    class="MuiTypography-root MuiTypography-h2 MuiTypography-alignCenter css-icizyq-MuiTypography-root"
  >
    5882
  </h2>,
  "requiredCal": <h2
    class="MuiTypography-root MuiTypography-h2 MuiTypography-alignCenter css-icizyq-MuiTypography-root"
  >
    0
  </h2>,
}
`;
